File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int l = 0, r = s.length() - 1;
        for(; l < r; l++, r--)
        {
            while(l < r and !isalnum(s[l]))
                l++;
            while(r > l and !isalnum(s[r]))
                r--;  
            if(char(tolower(s[l])) != char(tolower(s[r])))
                return false;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.cs
public class Solution {
     public bool IsPalindrome(string s) {
        int front = 0, back = s.Length - 1;

        s = s.ToLower();
        while (front < back)
        {
            while (front < back & !Char.IsLetterOrDigit(s[front]))
                front++;
            while (front < back & !Char.IsLetterOrDigit(s[back]))
                back--;
            if (s[front] != s[back])
                return false;
            front++;
            back--;
        }
        return true;
    }
}